src/_pytest/config/__init__.py

```diff
@@ -1104,10 +1108,11 @@ def inipath(self) -> pathlib.Path | None:
         """
         return self._inipath
 
-    def add_cleanup(self, func: Callable[[], None]) -> None:
+    def add_cleanup(self, func: _T_callback) -> _T_callback:
         """Add a function to be called when the config object gets out of
         use (usually coinciding with pytest_unconfigure)."""
```

Comment:
```suggestion
        use (usually coinciding with pytest_unconfigure).
        
        Returns the passed function.
        """
```

Besides that, can you explain why you changed the return type here?

Reply:
Oh it's just so it's nicer when used as a decorator